RFC6265bis: Gen-ART review - Address nits and editorial changes

sbingler opened the pull request:
This PR address many of the suggested nit and editorial changes with the [Gen-ART review](https://lists.w3.org/Archives/Public/ietf-http-wg/2025JanMar/0124.html).

It also cleans up a few other parts I found.

mikewest commented on draft-ietf-httpbis-rfc6265bis.md:
```diff
@@ -167,6 +142,31 @@ informative:
       ins: C. Jackson
       name: Collin Jackson
     date: 2011
+  HTML:
+    target: https://html.spec.whatwg.org/
+    title: HTML Living Standard
+    author:
+    -
+      ins: I. Hickson
+      name: Ian Hickson
+      organization: Google, Inc.
+    -
+      ins: S. Pieters
+      name: Simon Pieters
+      organization: Opera
+    -
+      ins: A. van Kesteren
+      name: Anne van Kesteren
+      organization: Mozilla
+    -
+      ins: P. Jägenstedt
+      name: Philip Jägenstedt
+      organization: Opera
+    -
+      ins: D. Denicola
+      name: Domenic Denicola
+      organization: Google, Inc.
+    ann: WHATWG
```
1. Specref suggests a slightly different set of editors (https://www.specref.org/?q=HTML%20Standard).
2. Anne is now at Apple, and Simon is now at Mozilla.
3. I'm unsure of Ian's current affiliation. He's not at Google anymore (https://ln.hixie.ch/?start=1700627373&count=1). 

```suggestion
  HTML:
    target: https://html.spec.whatwg.org/
    title: HTML Living Standard
    author:
    -
      ins: A. van Kesteren
      name: Anne van Kesteren
      organization: Apple
    -
      ins: D. Denicola
      name: Domenic Denicola
      organization: Google, Inc.
    -
      ins: D. Farolino
      name: Domenic Farolino
      organization: Google, Inc.
    -
      ins: I. Hickson
      name: Ian Hickson
    -
      ins: P. Jägenstedt
      name: Philip Jägenstedt
      organization: Opera
    -
      ins: S. Pieters
      name: Simon Pieters
      organization: Mozilla
    ann: WHATWG
```

sbingler commented:
Seems other spec drafts just don't include organizations, along with the specref you linked to, so I removed them from here as well. Also moved Fetch since it suffers from the same non-normative issues as HTML

PTAL

mikewest reviewed (approved):
Still LGTM, thanks.
